drivers/counter/counter_nrfx_rtc.c

```diff
@@ -25,17 +25,24 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME, LOG_LEVEL);
 
 #define COUNTER_MAX_TOP_VALUE RTC_COUNTER_COUNTER_Msk
 
-#define CC_TO_ID(cc) ((cc) - 1)
-#define ID_TO_CC(id) ((id) + 1)
+#define COUNTER_GET_TOP_CH(dev) counter_get_num_of_channels(dev)
 
-#define TOP_CH 0
-#define COUNTER_TOP_INT NRFX_RTC_INT_COMPARE0
-#define COUNTER_TOP_INT_MASK NRF_RTC_INT_COMPARE0_MASK
+#define IS_FIXED_TOP(dev) COND_CODE_1(CONFIG_COUNTER_RTC_CUSTOM_TOP_SUPPORT, \
+		(get_nrfx_config(dev)->fixed_top), (true))
+
+#define IS_PPI_WRAP(dev) COND_CODE_1(CONFIG_COUNTER_RTC_WITH_PPI_WRAP, \
+		(get_nrfx_config(dev)->use_ppi), (false))
+
+#define CC_ADJUSTED_OFFSET 16
+#define CC_ADJ_MASK(chan) (BIT(chan + CC_ADJUSTED_OFFSET))
 
 struct counter_nrfx_data {
 	counter_top_callback_t top_cb;
 	void *top_user_data;
 	u32_t top;
+	u32_t guard_period;
+	/* Store channel interrupt pending and CC adjusted flags. */
+	atomic_t ipend_adj;
```

Comment:
Nice and creative name, with a bit of a mystery.
No need to change it, I just wanted to express my admiration. :smile: